snuba/web/rpc/common/aggregation.py

```diff
@@ -0,0 +1,280 @@
+from dataclasses import dataclass
+from typing import Optional
+
+from sentry_protos.snuba.v1.trace_item_attribute_pb2 import (
+    AttributeAggregation,
+    ExtrapolationMode,
+    Function,
+)
+
+from snuba.query.dsl import CurriedFunctions as cf
+from snuba.query.dsl import Functions as f
+from snuba.query.dsl import column, literal
+from snuba.query.expressions import (
+    CurriedFunctionCall,
+    Expression,
+    FunctionCall,
+    SubscriptableReference,
+)
+from snuba.web.rpc.common.common import attribute_key_to_expression
+from snuba.web.rpc.common.exceptions import BadSnubaRPCRequestException
+
+sampling_weight_column = column("sampling_weight")
+
+# Z value for 95% confidence interval is 1.96 which comes from the normal distribution z score.
+z_value = 1.96
+
+CUSTOM_COLUMN_PREFIX = "__snuba_custom_column__"
+
+
+@dataclass(frozen=True)
+class CustomColumnInformation:
+    """
+    In order to support extrapolation, we need to be able to create a new column in clickhouse that computes some value, potentially based on some existing column.
+    This class holds the information needed to generate alias for the column so we can know what the column represents when getting results.
+    """
+
+    column_type: str
+    referenced_column: Optional[str]
+    metadata: dict[str, str]
+
+
+def _generate_custom_column_alias(
+    custom_column_information: CustomColumnInformation,
+) -> str:
+    alias = CUSTOM_COLUMN_PREFIX + custom_column_information.column_type
+    if custom_column_information.referenced_column is not None:
+        alias += f"${custom_column_information.referenced_column}"
+    if custom_column_information.metadata:
+        alias += "$" + ",".join(
+            [
+                f"{key}:{value}"
+                for key, value in custom_column_information.metadata.items()
+            ]
+        )
+    return alias
+
+
+def get_custom_column_information(alias: str) -> CustomColumnInformation:
+    if not alias.startswith(CUSTOM_COLUMN_PREFIX):
+        raise ValueError(f"Alias {alias} does not start with {CUSTOM_COLUMN_PREFIX}")
+
+    alias = alias[len(CUSTOM_COLUMN_PREFIX) :]
+    parts = alias.split("$")
+    column_type = parts[0]
+    referenced_column = parts[1] if len(parts) > 1 else None
+    metadata_parts = parts[2].split(",") if len(parts) > 2 else []
+    metadata = {}
+    for metadata_part in metadata_parts:
+        key, value = metadata_part.split(":")
+        metadata[key] = value
+
+    return CustomColumnInformation(column_type, referenced_column, metadata)
```

Comment:
style: these functions should be on the `CustomColumnInformation` class. Something like `to_alias` from_alias`